auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepo userRepo;
+    //private final RoleRepo roleRepo;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final MailService mailService;
+    private final JwtTokenUtil jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    @Value("${gateway.origin}")
+    private String gatewayOrigin;
+    @Value("${gateway.subdomain}")
+    private String gatewaySubdomain;
+    @Value("${gateway.protocol}")
+    private String gatewayProtocol;
+
+    @Override
+    public User getUserById(long id) {
+        if (id <= 0){
+            throw new InvalidInputException("Id cannot be a negative number for " + id);
+        }
+        else {
+            User entity  = userRepo.findById(id)
+                    .orElseThrow(() -> new NotFoundException("No user found for userID " + id));
+            log.info("User getUserById: found userId: {}", entity.getId());
+            return entity;
+        }
+    }
+
+    @Override
+    public Page<User> findAll(PageRequest of) {
+        return userRepo.findAll(of);
+    }
+
+    @Override
+    public List<User> findAllWithoutPage() {
+        return userRepo.findAll();
+    }
+
+    @Override
+    public User createUser(@Valid UserIDLessRoleLessDTO userIDLessDTO) {
+
+        final Optional<User> byEmail = userRepo.findByEmail(userIDLessDTO.getEmail());
+
+        if(byEmail.isPresent()) {
+            throw new EmailAlreadyExistsException(
+                    format("User with e-mail %s already exists", userIDLessDTO.getEmail()));
+        }
+
+        log.info("Saving user with email {}", userIDLessDTO.getEmail());
+        User user = userMapper.idLessRoleLessDTOToModel(userIDLessDTO);
+
+        //Optional<Role> role = roleRepo.findById(1L);
+        Set<Role> roleSet = new HashSet<>();
+        //role.ifPresent(roleSet::add);
+        user.setRoles(roleSet);
+
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+
+        log.info("Sending email to {}...", userIDLessDTO.getEmail());
+        log.info(mailService.sendMail(generateVerificationMail(user)));
+        log.info("Email sent to {}", userIDLessDTO.getEmail());
+
+        return userRepo.save(user);
+    }
+
+    @Override
+    public User passwordReset(long userId, @Valid String newPassword) {
+
+        log.info("id={}", userId);
+        User user = userRepo.findById(userId).orElseThrow(() -> new NotFoundException("No user for id:" + userId));
+        user.setPassword(newPassword);
+        return userRepo.save(user);
+
+    }
+
+    @Override
+    public void deleteUser(long userId) {
+        log.info("deleteUser: trying to delete entity with userId: {}", userId);
+        userRepo.findById(userId).ifPresent(userRepo::delete);
+    }
+
+    @Override
+    public Mail generateVerificationMail(User user) {
```

Comment:
How does the verification mail code work?

Reply:
For now it does not, at least not really